### ISSUE TITLE
Schema: record flag support for browser releases

### ISSUE THREAD

**Issue by ddbeck** [mdn/browser-compat-data#10616]
Several of the browsers we track don't have meaningful support for flags, including Opera for Android, Samsung Internet, Firefox for Android (from version 79), and Android WebView. If we recorded this support explicitly in BCD, we could lint against incorrectly-introduced flag data and increase the accuracy browser data mirroring.

## Proposal: add an `accepts_flags` key in browser release objects

I'm not set on the name, but I think it would be good to add a boolean property to browser release objects. It could have the following semantics:

- `true` would indicate that the browser release does have some notion of flags and support statements covering that release can contain flags data
- `false` would indicate that the browser does _not_ support flags in that release and that support statements that cover that release cannot contain flags data
- an omitted key would be equivalent to a `null` value: it's not known whether that release supports any flags

#### A browser that forbids flags

```
{
  "release_date": "2020-07-28",
  "release_notes": "https://developer.mozilla.org/docs/Mozilla/Firefox/Releases/79",
  "status": "retired",
  "engine": "Gecko",
  "engine_version": "79"
  "accepts_flags": false
}
```

#### A browser that allows flags

```
{
  "release_date": "2020-07-28",
  "release_notes": "https://developer.mozilla.org/docs/Mozilla/Firefox/Releases/79",
  "status": "retired",
  "engine": "Gecko",
  "engine_version": "67"
  "accepts_flags": true
}
```

#### A browser with unknown flag support

The key is omitted:

```
{
  "release_date": "2020-09-16",
  "release_notes": "https://developer.apple.com/documentation/safari-release-notes/safari-14-release-notes",
  "status": "current",
  "engine": "WebKit",
  "engine_version": "610.1.28"
}
```

Optionally, we could allow an explicit `null` value:

```
{
  "release_date": "2020-09-16",
  "release_notes": "https://developer.apple.com/documentation/safari-release-notes/safari-14-release-notes",
  "status": "current",
  "engine": "WebKit",
  "engine_version": "610.1.28"
  "accepts_flags": null
}
```

## Related issues

https://github.com/mdn/browser-compat-data/issues/4753
https://github.com/mdn/browser-compat-data/issues/7354
https://github.com/mdn/browser-compat-data/issues/9679

**Comment by ddbeck** [comment]
One additional thought on this: if we introduced this, we could retroactively change the `accepts_flag` value based on the release date, making it possible to at least partially automate the removal of irrelevant flag data.

**Comment by vinyldarkscratch** [comment]
+1 for this!  I briefly talked about the idea of having a property in the browser data in #5737, though the intention was more for the entire browser, rather than just a browser release.

**Comment by Elchi3** [comment]
Same feedback from me: I was expecting this to be on the browser level instead of on the release level.

I forgot the linter already does this: https://github.com/mdn/browser-compat-data/blob/main/test/linter/test-versions.js#L29

**Comment by vinyldarkscratch** [comment]
Thinking on this a little more, I can see a benefit to having release-level data of when flags are supported, as in the case of Firefox Android when flags were removed in Fenix.  However, since we're working to remove old flags from our data as it is, I'm thinking that it will ￼eventually become redundant, and keeping the data browser-level will be sufficient.

I like the idea of placing the data in the schema, as it allows for consumers to access this data as well, for whatever reason they might have.  One thought I had was: why don't we base whether a browser supports flags based on the presence of `pref_url`?

**Comment by ddbeck** [comment]
> why don't we base whether a browser supports flags based on the presence of `pref_url`?

I considered that, but Safari would be left out. It has preferences, but they're provided through menus, not a special URL.

**Comment by ddbeck** [comment]
Fixed by #11286.
